### ISSUE TITLE
Fix bad helm style - tech debt

### ISSUE THREAD

**Issue by kdorosh** [solo-io/gloo#2705]
Was writing upgrade docs, realized just how painful our helm charts and related docs are to use.

User pain:
1. Finding documentation of enterprise helm chart values, which is always out of date, because it isn't auto-generated. this is especially bad because we link to it in several places in our docs as a "complete" list of helm values. also, this list should contain the _enterprise specific_ values only
1. Distinction between values that need gloo prefix vs not needing the prefix (e.g., enterprise settings vs all other helm values). needing the prefix at all is painful during upgrades
1. tons of version-specific instructions to get around previous helm chart bugs in Gloo (e.g., the majority of our upgrade guides). we should test and ensure `helm upgrade` always works, simple as that.
1. documentation bloat -- caused by documenting around the above the issues, when we could do more with less
1. not exposing all configuration settings in our helm charts

Dev Pain:
1. Difficult to ensure all options are exposed in settings in the helm charts because settings is duplicated across the Gloo and Gloo Enterprise chart
1. our helm tests are largely copy-paste, and terrible to maintain
1. we need more helm regression tests before we can attempt the refactor we need to do
1. bugs in helm chart common because we don't automate testing enough paths of provided helm values (e.g., custom gateway proxies leading to us dereferencing nils)

To fix:
- [ ] add regression testing in gloo enterprise for `helm upgrade`
- [ ] de-deduplicate helm values by trimming down values.go in Gloo Enterprise to the bare necessities
- [ ] use the same helm chart value docs gen logic in Gloo Enterprise that we use in Gloo
- [ ] use the go objects to generate the template overrides, or allow for custom string yaml on every object spec, so we can't forget to expose a new setting in a helm chart
- [ ] once well tested and simplified, we can begin to deprecate and remove docs instructions for older versions of gloo, relegating them to the older hosted versions of the docs, removing bloat
- [ ] enforce consistency by making everything (including settings) become nested underneath gloo (except for globals of course) so we don't have to document warnings all over our guides depending on your version

Understood this is a laundry list of issues and that some of them are duplicates of older issues, but reopening so we can better split out work and prioritize this, as the problem has only gotten worse since last raised.

**Comment by yuval-k** [comment]
one thought i had for consistency, is to have most of gloo be a subchart of gloo. this way gloo-e can pull the same subchart, and values will look exactly the same

**Comment by kdorosh** [comment]
Another idea is to use helm3 import/exported values from dependencies https://helm.sh/docs/topics/charts/#importing-child-values-via-dependencies

This only cleans up our helm3 use case, however. This is not an option in helm2
